doc/man7/provider-base.pod

```diff
@@ -156,7 +156,7 @@ provider):
  {BIO_free}                       OSSL_FUNC_BIO_FREE
  {BIO_vprintf}                    OSSL_FUNC_BIO_VPRINTF
  {OPENSSL_cleanse}                OSSL_FUNC_OPENSSL_CLEANSE
- {OSSL_SELF_TEST_set_callback}    OSSL_FUNC_SELF_TEST_CB
+ {self_test_cb}                   OSSL_FUNC_SELF_TEST_CB
```

Comment:
We could also call it `SELF_TEST_set_callback` (without OSSL_) which corresponds to 

- OSSL_FUNC_SELF_TEST_SET_CALLBACK
- OSSL_FUNC_SELF_TEST_set_callback()

Reply:
This set of functions is a bit special, ass they are upcalls that are direct pointers to public OpenSSL functions of the exact same name.  For example, the one just above, `OPENSSL_cleanse`, is an actual OpenSSL function.  So, following the same scheme with this callback setter, it should be:

```
{OSSL_SELF_TEST_set_callback}    OSSL_FUNC_OSSL_SELF_TEST_SET_CALLBACK
```

Reply:
I understand that, but we are not forced to name the fetched upcall _exactly_ the same way as the original function. The old types `CRYPTO`, `BIO`, etc. are different, they did not have an `SSL_` prefix enforced yet, contrary to `SELF_TEST`. Do we really want those prefixes to add up?

The helper function would become

```
OSSL_FUNC_OSSL_SELF_TEST_set_callback()

```
which looks a bit ridiculous, instead of

```
OSSL_FUNC_SELF_TEST_set_callback()
```

which would be sufficient IMHO. So I would prefer naming it

```
{SELF_TEST_set_callback}    OSSL_FUNC_SELF_TEST_SET_CALLBACK
```

Reply:
Thinking in C++, the `SSL_` prefix is a global namespace, so it shouldn't accumulate.